orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db.sh

```diff
@@ -27,26 +27,71 @@ doLog() {
   echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg ""$msg" >>$LOGFILE
 }
 
-if [[ $# -lt 6 || $# -gt 10 || "$1" == "None" ]]; then
-  doLog "Invalid inputs provided"
-  doLog "Script accepts at least 6 and at most 7 inputs:"
-  doLog "  1. Object Storage Service url to place backups."
-  doLog "  2. PostgreSQL host name."
-  doLog "  3. PostgreSQL port."
-  doLog "  4. PostgreSQL user name."
-  doLog "  5. Ranger admin group."
-  doLog "  6. Whether or not to close connections for the database while it is being backed up."
-  doLog "  7-10. (optional) Names of the databases to backup. If not given, will backup ranger and hive databases."
-  exit 1
-fi
+log_invalid_input_reason() {
+  doLog "This might be caused by the command is not running on the master Getaway node or there has never had a backup/restore from the CDP CLI run."
+}
 
-BACKUP_LOCATION="$1"
-HOST="$2"
-PORT="$3"
-USERNAME="$4"
-RANGERGROUP="$5"
-CLOSECONNECTIONS="$6"
-DATABASENAMES="${@: 7}"
+# Print out all required inputs if they are passed in.
+while getopts ":s:h:p:u:g:c:n:" OPTION; do
+    case $OPTION in
+    s  )
+        STORAGE_URL="$OPTARG"
+        doLog "Object Storage Service url to place backups: $STORAGE_URL"
+        ;;
+    h  )
+        HOST="$OPTARG"
+        doLog "PostgreSQL host name: $HOST"
+        ;;
+    p  )
+        PORT="$OPTARG"
+        doLog "PostgreSQL port: $PORT"
+        ;;
+    u  )
+        USERNAME="$OPTARG"
+        doLog "PostgreSQL user name: $USERNAME"
+        ;;
+    g  )
+        GROUP="$OPTARG"
+        doLog "Ranger admin group: $GROUP"
+        ;;
+    c  )
+        CLOSE_CONNECT="$OPTARG"
+        doLog "Whether or not to close connections for the database while it is being backed up: $CLOSE_CONNECT"
+        ;;
+    n  )
+        DB_NAME+=("$OPTARG")
+        DB_NAME_ITEM="$OPTARG"
+        doLog "Names of the databases to backup. If not given, will backup ranger and hive databases: $DB_NAME_ITEM"
+        ;;
+    \? ) doLog "Unknown option: -$OPTARG" >&2;
+        exit 1;;
+    :  ) echo "Missing option argument for -$OPTARG" >&2;
+        log_invalid_input_reason
+        exit 1;;
+    esac
+done
+
+# If any of the required inputs are missing or the provided values are NULL, the script can not run successfully.
+[ -z ${STORAGE_URL+x} ] || [ -z ${GROUP+x} ] || [ -z ${CLOSE_CONNECT+x} ]||
+[ -z "$STORAGE_URL" ] || [ -z "$GROUP" ] || [ -z "$CLOSE_CONNECT" ] &&
+doLog "At least one mandatory parameter is not set in disaster_recovery.sls: Object Storage Service URL; Ranger admin group; close connections or not during backup." && log_invalid_input_reason && exit 1
+[ -z ${HOST+x} ] || [ -z ${PORT+x}  ] || [ -z ${USERNAME+x} ] ||
+[ -z "$HOST" ] || [ -z "$PORT" ] || [ -z "$USERNAME" ] &&
```

Comment:
In `backup.sls` we see that the script can be called with 

```
/opt/salt/scripts/backup_db.sh{% if storage %} -s {{storage}}{% endif %} -h "" -p "" -u ""{% if group %} -g {{group}}{% endif %}{% if dbname %}{% for item in dbname_list %} -n {{item}}{% endfor %}{% endif %}
```
So there is a case where we run the script with `-h "" -p "" -u ""` so this would prevent backup/restore in that case. 

However, this case is only for when an external DB is not used which isn't the case for any customer, only some internal uses. I recommend confirming with Kalyan that we still want to support this case.

Reply:
@sxxgrc Sure! I will ask Kalyan about it. The added `-h` `-p` shouldn't impact how it functions though since the passed in paramaters are still empty. I tested this and it can not work, which is expected. When I tested it I don't know what is the purpose of this cmd though. Thank you for letting me know!